src/components/livestore/LivestoreProviderProvider.tsx

```diff
@@ -0,0 +1,167 @@
+import { LiveStoreProvider } from "@livestore/react";
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { unstable_batchedUpdates as batchUpdates } from "react-dom";
+import { useAuth } from "../../auth/index.js";
+import LiveStoreWorker from "./livestore.worker?worker";
+
+interface BetterLiveStoreContextType {
+  isReady: boolean;
+}
+
+const BetterLiveStoreContext = createContext<
+  BetterLiveStoreContextType | undefined
+>(undefined);
+
+function loading(_status: BootStatus) {
+  // Let our overlay handle loading
+  // console.debug(`LiveStore Loading status: ${JSON.stringify(_status)}`);
+  return <></>;
+}
+
+// Component to detect when LiveStore is ready
+const LiveStoreReadyDetector: React.FC<{ onReady?: () => void }> = ({
+  onReady,
+}) => {
+  const readyRef = useRef(false);
+
+  useEffect(() => {
+    // If this component renders, LiveStore is ready
+    if (!readyRef.current) {
+      readyRef.current = true;
+      onReady?.();
+    }
+  }, [onReady]);
+
+  return null;
+};
+
+interface LiveStoreProviderProviderProps {
+  storeId: string;
+  onLiveStoreReady?: () => void;
+  children: React.ReactNode;
+}
+
+// LiveStore setup - moved inside AuthGuard to ensure auth happens first
+export const LiveStoreProviderProvider: React.FC<
+  LiveStoreProviderProviderProps
+> = ({ storeId, onLiveStoreReady, children }) => {
+  const [liveStoreReady, setLiveStoreReady] = useState(false);
+
+  // Check for reset parameter to handle schema evolution issues
+  const resetPersistence =
+    new URLSearchParams(window.location.search).get("reset") !== null;
+
+  // Clean up URL if reset was requested
+  useEffect(() => {
+    if (resetPersistence) {
+      const searchParams = new URLSearchParams(window.location.search);
+      searchParams.delete("reset");
+      window.history.replaceState(
+        null,
+        "",
+        `${window.location.pathname}?${searchParams.toString()}`
+      );
+    }
+  }, [resetPersistence]);
+
+  // Get authenticated user info and access token
+  const {
+    user: { sub: clientId },
+    accessToken,
+  } = useAuth();
+
+  // Create completely static sync payload that never changes reference
+  // Token and clientId are updated via useEffect to prevent LiveStore restarts
+  const syncPayload = useRef({
+    authToken: accessToken || "",
+    clientId,
+  });
+
+  // Update clientId and authToken if they change, but keep same object reference
+  useEffect(() => {
+    syncPayload.current.clientId = clientId;
+    syncPayload.current.authToken = accessToken || "";
+  }, [clientId, accessToken]);
+
+  const adapter = useMemo(
+    () =>
+      makePersistedAdapter({
+        storage: { type: "opfs" },
+        worker: LiveStoreWorker,
+        sharedWorker: LiveStoreSharedWorker,
+        resetPersistence,
+        clientId, // This ties the LiveStore client to the authenticated user
+      }),
+    [clientId, resetPersistence]
+  );
```

Comment:
Once you bring us back to _zero_ `clientId` being passed in, I think we can now finally move the adapter up to be a singleton in this module. We'd handle resetPersistence on the outside of the component here (if we even keep it).